packages/@lwc/engine-server/src/types.ts

```diff
@@ -14,45 +19,45 @@ export enum HostNodeType {
 }
 
 export interface HostText {
-    type: HostNodeType.Text;
-    parent: HostElement | null;
+    [HostTypeKey]: HostNodeType.Text;
+    [HostParentKey]: HostElement | null;
     value: string;
 }
 
 export interface HostComment {
-    type: HostNodeType.Comment;
-    parent: HostElement | null;
+    [HostTypeKey]: HostNodeType.Comment;
+    [HostParentKey]: HostElement | null;
     value: string;
 }
 
 export interface HostRaw {
-    type: HostNodeType.Raw;
-    parent: HostElement | null;
+    [HostTypeKey]: HostNodeType.Raw;
+    [HostParentKey]: HostElement | null;
     value: string;
 }
 
 export interface HostAttribute {
     name: string;
-    namespace: string | null;
+    [HostNamespaceKey]: string | null;
     value: string;
 }
 
 export interface HostShadowRoot {
-    type: HostNodeType.ShadowRoot;
+    [HostTypeKey]: HostNodeType.ShadowRoot;
     children: HostChildNode[];
     mode: 'open' | 'closed';
     delegatesFocus: boolean;
 }
 
 export interface HostElement {
-    type: HostNodeType.Element;
-    name: string;
-    namespace: string;
-    parent: HostElement | null;
+    [HostTypeKey]: HostNodeType.Element;
```

Comment:
@pmdartus Expanded the Symbols to cover the other internal properties.

I left `shadowRoot`, `children`, and `attributes` open because those are actual HTML properties inherited from Element and not purely internal properties. Will changing them in this way break the existing usage of those properties? Do we _want_ users to be able to use them as public props in this way?

Reply:
FYI for anyone else, the ask here was to harden our implementation for public props to not conflict with HostElement internal properties.

See comment at https://github.com/salesforce/lwc/pull/2934/files#r924208524

Reply:
It's a really good question and I took some time to think about this today. I came to the following conclusion that all the host element's internal properties should be declared as symbols.

Our mock implementation of the DOM is quite naive for SSR. In a real browser, all the DOM nodes are backed by a C++ object storing the internal properties of the node. For instance, redefining the `attributes` property on an element or its prototype would not have any impact on` getAttribute` or `setAttribute`. 

That said, nothing prevents developers from redefining a property or a method present on the `HTMLElement` prototype. For instance, if a component redefines `insertBefore`, the component would fail to mount since this method is used by the diffing algo. 

I think moving all the internal properties of a symbol would be the safest approach here.